app/src/main/java/com/moxielabs/androidarchitectureexamples/shared/models/Fruit.kt

```diff
@@ -4,4 +4,8 @@ data class Fruit(
     val name: String,
     val color: String,
     val taste: String
-)
\ No newline at end of file
+) {
+    fun displayText(): String {
```

Comment:
This isn't something that we'd usually put in the model for MVVM. If we're adding something like displayText and MVP and MVC would both normally have that type of function inside the Model, then I can make a standalone model for MVVM and put the displayText stuff where it should be for MVVM

Reply:
I hadn't thought of it from that angle. This is an old habit for me. I favor this pattern over creating an extension function. Which is my other option. What would you prefer?

Reply:
or does it make more sense for this pattern to not share this model?

Reply:
My understanding is that in MVP, the `Presenter` handles manipulating what the `Model` is holding and getting it ready to present to the `View`. I also understand that it's acceptable for the `Model` to contain business logic in MVP, so I'm not sure there's a clear space where this kind of method should be for MVP. Not that I can tell, at least.

So I think it's fine to keep the text manipulation/preparing in either the `Model` like you have it or the `Presenter`. I haven't put thought yet into whether MVI or Clean would store that type of function in the model.

I don't know where data manipulation from the model values occurs in MVC.

I don't think it matters much whether we make a separate `Model` for MVP/MVC or we make a separate `Model` for the archs that don't put those function types in the `Model`. So I'm arbitrarily picking a default of sharing a `Model` if its just model properties inside it, and creating an arch-specific model if that arch shoves more things into the model.

Reply:
Makes sense. It's funny to have to think of these things as separate and together at the same time. I'm going to create extension functions separately for MVC and MVP. MVC function will live in the fragment. MVP function will live in presenter. I think that would maybe even highlight how design pattern can affect usages.